warehouse/ingest-core/src/main/java/datawave/ingest/data/config/ingest/VirtualIngest.java

```diff
@@ -383,11 +405,58 @@ private void compilePatterns() {
             compiledFieldPatterns = patterns;
         }
         
+        public class VirtualFieldGrouping {
+            private final String group0;
+            private final String group1;
+            
+            public VirtualFieldGrouping(String group0, String group1) {
+                this.group0 = group0;
+                this.group1 = group1;
+            }
+            
+            public String getGroup1() {
+                return group1;
+            }
+            
+            public String getGroup0() {
+                return group0;
+            }
+            
+            public String get(int group) {
+                if (group == 0) {
+                    return group0;
+                } else if (group == 1) {
+                    return group1;
+                } else {
+                    throw new IllegalArgumentException("Illegal group " + group);
+                }
+            }
+            
+            @Override
+            public int hashCode() {
+                return new HashCodeBuilder().append(group0).append(group1).toHashCode();
+            }
+            
+            @Override
+            public boolean equals(Object obj) {
+                if (obj instanceof VirtualFieldGrouping) {
+                    VirtualFieldGrouping other = (VirtualFieldGrouping) obj;
+                    return new EqualsBuilder().append(group0, other.group0).append(group1, other.group1).isEquals();
+                }
+                return false;
+            }
+            
+            @Override
+            public String toString() {
+                return group0 + '.' + group1;
+            }
+        }
+        
         // Create the virtual field from the event Fields members
         public void addVirtualFields(List<NormalizedContentInterface> virtualFields, Multimap<String,NormalizedContentInterface> eventFields,
```

Comment:
It might help (me, at least) if there were some comments for this method that describe how we're going about adding the appropriate fields.